sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/io/BigQueryIO.java

```diff
@@ -1104,13 +1104,17 @@ public void finalize(Iterable<FileResult> writerResults, PipelineOptions options
           tempFiles.add(result.getFilename());
         }
         if (!tempFiles.isEmpty()) {
-          inserter.load(
-              bigQuerySink.jobIdToken,
-              JSON_FACTORY.fromString(bigQuerySink.jsonTable, TableReference.class),
-              tempFiles,
-              JSON_FACTORY.fromString(bigQuerySink.jsonSchema, TableSchema.class),
-              bigQuerySink.writeDisposition,
-              bigQuerySink.createDisposition);
+          try {
+            inserter.load(
+                bigQuerySink.jobIdToken,
+                JSON_FACTORY.fromString(bigQuerySink.jsonTable, TableReference.class),
+                tempFiles,
+                JSON_FACTORY.fromString(bigQuerySink.jsonSchema, TableSchema.class),
+                bigQuerySink.writeDisposition,
+                bigQuerySink.createDisposition);
+          } finally {
+            removeTemporaryFiles(options);
```

Comment:
This is how the the default FileBasedSink.finalize() doing the cleanup. I think BigQueryIO should do the same as TextIO.
If you think it is wrong, I can write a separate PR to change it to:
removeTemporaryFiles(Iterable<FileResult> writerResults).